### PR TITLE
docs(blog): introducing PWA

### PR DESCRIPTION
**Pull request by KaiVandivier**

### CODE REVIEW

File: blog/2021-11-04-introducing-pwa.md
Comment (mediremi) [inline blog/2021-11-04-introducing-pwa.md:34]:
```suggestion
To use the basic PWA features (offline caching and installability) in an app, simply opt-in to PWA using an option in [`d2.config.js`](https://platform.dhis2.nu/#/config/d2-config-js-reference):
```

**Review by mediremi — APPROVED** [review]
LGTM :tada: 

Covers every aspect of PWA imo

File: blog/2021-11-04-introducing-pwa.md
Comment (amcgee) [inline blog/2021-11-04-introducing-pwa.md:79]: minor: we already use the term **app shell** to refer to the common wrapper provided by the platform.  Using this terminology again to refer to something a bit different might be confusing...

File: blog/2021-11-04-introducing-pwa.md
Comment (amcgee) [inline blog/2021-11-04-introducing-pwa.md:166]:
```suggestion
In the near future we will also post a detailed article about the technical development of these PWA features because we’re using these tools in a pretty unique and cool way that we want to share, so keep an eye out for that post!
```

**Review by amcgee — CHANGES_REQUESTED** [review]
Looks great!!  Minor comments, let's run this by the @germanviscuso and the security team before posting

File: blog/2021-11-04-introducing-pwa.md
Comment (varl) [inline blog/2021-11-04-introducing-pwa.md:6]:
```suggestion
author_image_url: https://github.com/KaiVandivier.png
```

File: blog/2021-11-04-introducing-pwa.md
Comment (varl) [inline blog/2021-11-04-introducing-pwa.md:10]: I noticed that "Dashboards App" and "Dashboard App" (L108) are both used in this article, and "Dashboard" (singular) is preferred:

- https://github.com/dhis2/dashboard-app
- https://github.com/dhis2/dashboard-app/blob/master/package.json#L2
- https://github.com/dhis2/dashboard-app/blob/master/package.json#L4
- https://github.com/dhis2/dashboard-app/blob/master/d2.config.js#L4

Historically we did have a "Dashboards" app, but that was only because the name "Dashboard" was already in use. This was solved a while ago and now we only have the "Dashboard" app.

File: blog/2021-11-04-introducing-pwa.md
Comment (varl) [reply blog/2021-11-04-introducing-pwa.md:79]: I was thinking that we can avoid giving it a name and gloss over it a bit:

```suggestion
The dashboard here is the **cacheable section** – it’s only made available offline when you do so explicitly. The rest of the app that continues to work while offline includes content like the scripts that run the app’s behavior, some user and server data, and the user’s list of dashboards.
```

**Review by varl — APPROVED** [review]
Very nice write up, @KaiVandivier. Good work !

Since we mention the Dashboard multiple times, maybe we can shoot a link to the GitHub repo if they are interested in checking out the actual implementation ?

File: blog/2021-11-04-introducing-pwa.md
Comment (varl) [inline blog/2021-11-04-introducing-pwa.md:2]:
```suggestion
slug: 2021/11/introducing-pwa
```

**Review by varl — CHANGES_REQUESTED** [review]
Change the slug to include the year/month before merging.

File: blog/2021-11-04-introducing-pwa.md
Comment (KaiVandivier) [reply blog/2021-11-04-introducing-pwa.md:10]: Gotcha, thanks!

File: blog/2021-11-04-introducing-pwa.md
Comment (KaiVandivier) [reply blog/2021-11-04-introducing-pwa.md:79]: Yeah it is regrettable to have that terminology conflict; I've been using "app shell" in the caching context because it seems like that's how people refer to "the static assets and data needed to run the core of the app" in PWA contexts. If you think we need a new name for it, we could work on a better one, but will have to rename a few [config options](https://github.com/dhis2/app-platform/blob/1d0423e135b71d2005198287075e47d939040049/cli/config/d2.pwa.config.js#L25) and change some documentation

What do you think about adding some more description to make it clearer that "app shell" here is referring to the cache contents, not to be confused with the common wrapper? Or removing the reference to "app shell" here like Viktor suggests?

**Review by varl — APPROVED** [review]

File: blog/2021-11-04-introducing-pwa.md
Comment (amcgee) [inline blog/2021-11-04-introducing-pwa.md:158]:
```suggestion
In its current implementation, **cached data is not encrypted** when stored offline, and a malicious actor could inspect access and inspect the cached data if they gain access to a user’s device. For this reason, it is **not recommended** to use PWA features in an app that will handle sensitive data, especially if the app is likely to be used on shared devices.  Some protections are currently provided by the platform to guard against unauthorized access to data when a user logs out or when a new user logs in, but without encryption it might still be possible for a savvy attacker to access some offline data.  Importantly, the data cached by the PWA is the same as might be cached at the browser layer in a non-PWA application, and so the risk of data exposure has not significantly increased.
```

**Review by amcgee — APPROVED** [review]
@kai looking good!  I made a final wording update here https://github.com/dhis2/developer-portal/pull/53/commits/a58c444794a9b2dd9ff982cab3dea41ee6ff04e1 thanks to a suggestion from Jaime - if you're ok with that let's go ahead and get this in.

**Comment by KaiVandivier** [comment]
Looks good, thanks!

**Pull request merged**
